src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -544,24 +544,24 @@ private static bool TryParseVersion(string version, ref VersionResult result)
 
             var versionSansLabel = (dashIndex < 0) ? version : version.Substring(0, dashIndex);
             string[] parsedComponents = versionSansLabel.Split(Utils.Separators.Dot);
-            if (parsedComponents.Length != 3)
+            if (parsedComponents.Length == 0 || parsedComponents.Length > 3)
```

Comment:
>arsedComponents.Length == 0  [](start = 17, length = 28)

i don't know if this condition can exist. Even `[string]::Empty.Split('.').Length will return 1